src/profile.c

```diff
@@ -487,13 +483,61 @@ static inline void printOutput(UInt line, int nameid, int exec, int visited)
     }
 }
 
+// Mark line as visited, and return true if the line has been previously
+// visited (executed)
+BOOL markVisited(int nameid, UInt line)
+{
+    if (nameid == 0 || line == 0) {
+        return 1;
+    }
+
+    if (LEN_PLIST(profileState.visitedStatements) < nameid ||
+        !ELM_PLIST(profileState.visitedStatements, nameid)) {
+        AssPlist(profileState.visitedStatements, nameid,
+                 NEW_PLIST(T_PLIST, 0));
+    }
+
+    Obj linelist = ELM_PLIST(profileState.visitedStatements, nameid);
+
+    if (LEN_PLIST(linelist) < line || !ELM_PLIST(linelist, line)) {
+        AssPlist(linelist, line, True);
+        return 0;
+    }
+    else {
+        return 1;
+    }
+}
+
+// Return TRUE is Stat has been visited (executed) before
+BOOL visitedStat(Stat stat)
+{
+    int nameid = getFilenameIdOfCurrentFunction();
+    int line = LINE_STAT(stat);
+
+    if (nameid == 0 || line == 0) {
+        return 1;
+    }
+
+    if (LEN_PLIST(profileState.visitedStatements) < nameid ||
+        !ELM_PLIST(profileState.visitedStatements, nameid)) {
+        return 1;
```

Comment:
Shouldn't this return *false*?
```suggestion
        return FALSE;
```